internal/pkg/cgroup/cgroup.go

```diff
@@ -56,3 +62,133 @@ func Path(cgroupPath string) string {
+			CPU: &cgroup2.CPU{
+				Weight: pointer.To[uint64](MillicoresToCPUWeight(MilliCores(constants.CgroupSystemMillicores))),
+			},
+		},
+		constants.CgroupSystemRuntime: {
+			Memory: &cgroup2.Memory{
+				Min: pointer.To[int64](constants.CgroupSystemRuntimeReservedMemory),
+				Low: pointer.To[int64](constants.CgroupSystemRuntimeReservedMemory * 2),
+			},
+			CPU: &cgroup2.CPU{
+				Weight: pointer.To[uint64](MillicoresToCPUWeight(MilliCores(constants.CgroupSystemRuntimeMillicores))),
+			},
+		},
+		constants.CgroupUdevd: {
+			Memory: &cgroup2.Memory{
+				Min: pointer.To[int64](constants.CgroupUdevdReservedMemory),
+				Low: pointer.To[int64](constants.CgroupUdevdReservedMemory * 2),
+			},
+			CPU: &cgroup2.CPU{
+				Weight: pointer.To[uint64](MillicoresToCPUWeight(MilliCores(constants.CgroupUdevdMillicores))),
+			},
+		},
+		constants.CgroupPodRuntimeRoot: {
+			CPU: &cgroup2.CPU{
+				Weight: pointer.To[uint64](MillicoresToCPUWeight(MilliCores(constants.CgroupPodRuntimeRootMillicores))),
+			},
+		},
+		constants.CgroupPodRuntime: {
+			Memory: &cgroup2.Memory{
+				Min: pointer.To[int64](constants.CgroupPodRuntimeReservedMemory),
+				Low: pointer.To[int64](constants.CgroupPodRuntimeReservedMemory * 2),
+			},
+			CPU: &cgroup2.CPU{
+				Weight: pointer.To[uint64](MillicoresToCPUWeight(MilliCores(constants.CgroupPodRuntimeMillicores))),
+			},
+		},
+		constants.CgroupKubelet: {
+			Memory: &cgroup2.Memory{
+				Min: pointer.To[int64](constants.CgroupKubeletReservedMemory),
+				Low: pointer.To[int64](constants.CgroupKubeletReservedMemory * 2),
+			},
+			CPU: &cgroup2.CPU{
+				Weight: pointer.To[uint64](MillicoresToCPUWeight(MilliCores(constants.CgroupKubeletMillicores))),
+			},
+		},
+		constants.CgroupDashboard: {
+			Memory: &cgroup2.Memory{
+				Max: zeroIfRace(pointer.To[int64](constants.CgroupDashboardMaxMemory)),
+			},
+			CPU: &cgroup2.CPU{
+				Weight: pointer.To[uint64](MillicoresToCPUWeight(MilliCores(constants.CgroupDashboardMillicores))),
+			},
+		},
+		constants.CgroupApid: {
+			Memory: &cgroup2.Memory{
+				Min:  pointer.To[int64](constants.CgroupApidReservedMemory),
+				Low:  pointer.To[int64](constants.CgroupApidReservedMemory * 2),
+				Max:  zeroIfRace(pointer.To[int64](constants.CgroupApidMaxMemory)),
+				Swap: pointer.To[int64](0),
+			},
+			CPU: &cgroup2.CPU{
+				Weight: pointer.To[uint64](MillicoresToCPUWeight(MilliCores(constants.CgroupApidMillicores))),
+			},
+		},
+		constants.CgroupTrustd: {
+			Memory: &cgroup2.Memory{
+				Min:  pointer.To[int64](constants.CgroupTrustdReservedMemory),
+				Low:  pointer.To[int64](constants.CgroupTrustdReservedMemory * 2),
+				Max:  zeroIfRace(pointer.To[int64](constants.CgroupTrustdMaxMemory)),
+				Swap: pointer.To[int64](0),
+			},
+			CPU: &cgroup2.CPU{
+				Weight: pointer.To[uint64](MillicoresToCPUWeight(MilliCores(constants.CgroupTrustdMillicores))),
+			},
+		},
+	}
+
+	inContainer := false
+	if _, err := os.Stat("/usr/etc/in-container"); err == nil {
```

Comment:
should we extract this into a single place, and use `sync.OnceValue` for it? 

we already have like 6 references to it

Reply:
needs to be done, what package should it be in? Machinery? Also perhaps needs to be constified

Reply:
certainly not machinery, as it's Talos-internal, it doesn't make sense outside of Talos, so could be `internal/containermode` ??

Reply:
Would make sense, will place under internal/ directly